lib/swoosh/adapters/zepto_mail.ex

```diff
@@ -0,0 +1,277 @@
+      |> cc("thor.odinson@example.com")
+      |> bcc({"Clinton Francis Barton", "hawk.eye@example.com"})
+      |> bcc("beast.avengers@example.com")
+      |> subject("Hello, Avengers!")
+      |> html_body("<h1>Hello</h1><img src=\\"cid:inline-attachment-from-cache\\" />")
+      |> text_body("Hello")
+      |> put_provider_option(:bounce_address, "bounce@example.com")
+      |> put_provider_option(:track_clicks, false)
+      |> put_provider_option(:track_opens, true)
+      |> put_provider_option(:inline_images, [%{cid: "inline-attachment-from-cache", file_cache_key: "cache-key"}])
+
+  ## Batch Sending
+
+  `ZeptoMail` does not support sending multiple different emails, however it does support sending one email
+  to a list of recipients.
+
+  To allow the customization of the email per recipient, a field `:merge_info` may be provided for each recipient.
+
+      import Swoosh.Email
+
+      new()
+      |> from({"T Stark", "tony.stark@example.com"})
+      |> to({"Steve Rogers", "steve.rogers@example.com"})
+      |> to("wasp.avengers@example.com")
+      |> subject("Hello, Avengers!")
+      |> html_body("<h1>Hello Avenger from {{ team }}</h1>")
+      |> put_provider_option(:merge_info,
+        %{
+          "steve.rogers@example.com" => %{team: "Avengers"},
+          "wasp.avengers@example.com" => %{team: "Avengers 2"}
+        }
+      )
+
+  ## Provider options
+    * `:bounce_address` (string) - The email address to which bounced emails will be sent.
+
+    * `:track_clicks` (boolean) - Enable or disable email click tracking.
+
+    * `:track_opens` (boolean) - Enable or disable email open tracking.
+
+    * `:client_reference` (string) - An identifier set by the user to track a particular transaction.
+
+    * `:mime_headers` (map) - The additional headers to be sent in the email for your reference purposes.
+
+    * `:attachments` (list) - A list of file cache keys to send as attachments.
+
+    * `:inline_images` (list) - A list of map (`cid` and `file_cache_key`) to include as inline attachments.
+
+    * `:template_key` (string) - Unique key identifier of your template.
+
+    * `:template_alias` (string) - Alias name given to the template key, can be used instead of `template_key`.
+
+    * `:merge_info` (map) - Use this values to replace the placeholders in the template.
+
+      In case of batch email sending, this should be a map of email address to a map of key-value.
+  """
+
+  use Swoosh.Adapter, required_config: [:api_key]
+
+  alias Swoosh.Email
+
+  @base_url "https://api.zeptomail.com/v1.1"
+  @api_endpoint "/email"
+  @api_template_endpoint "/email/template"
+
+  defguardp is_template(email)
+            when is_map_key(email.provider_options, :template_key) or
+                   is_map_key(email.provider_options, :template_alias)
+
+  def deliver(%Email{} = email, config \\ []) do
+    do_deliver(:single, email, config)
+  end
+
+  def deliver_many([email], config \\ []) do
```

Comment:
This only matches a list of one item. Probably not what you intended.

If intended, we'd rather not implement this function.

Reply:
I explained the reason behind this on the module documentation.
Sending batch email is a useful feature of ZeptoMail.  it works by sending one email to each recipient with an ability to customize variables for each one via `merge_info`.

If you think it's not how `deliver_many/2` is intended to be used, I'll try to make it as another option in the `config` field.

Reply:
I'll take a better look later. Cheers for the quick reply.

Reply:
Yes I think using the provider option would be more appropriate, instead of `deliver_many/2`.

Same templating functionality exists in many other providers.